dts/riscv32/rv32m1_ri5cy.dtsi

```diff
@@ -41,6 +41,21 @@
 		};
 	};
 
+	/*
+	 * Assumes ARM cores are disabled. Please note that these values are
+	 * a subject to change if SOC_OPEN_ISA_RV32M1_RI5CY_ZERO
+	 * is enabled in config.
```

Comment:
In moving memory definitions to dts, these values shouldn't change. We should instead define multiple flash regions, with the appropriate region being chosen based on the core selected.

I have a PR to enable the zero-riscy core in #15542.

Reply:
I agree. Zephyr does not currently support either of the ARM cores (or zero-risky, for that matter), but hopefully support for these will land in the tree soon.

Reply:
I splitted the device tree information into three files: base (the common one) and two specific to riscy and zero_riscy.

Reply:
@MaureenHelm Is the current solution ok?

Reply:
No, please see the last two commits in #15943

Reply:
I see that your PR is merged to master, so I rebased onto it and dropped my changes.

Reply:
@MaureenHelm I guess this closes the discussion, please confirm.

Reply:
Yes, thank you